src/core/AttestationManager.sol

```diff
@@ -0,0 +1,263 @@
+
+import { EMPTY_ATTESTATION_REF, EMPTY_RESOLVER_UID, _time, ZERO_TIMESTAMP } from "../Common.sol";
+
+/**
+ * AttestationManager handles the registry's internal storage of new attestations and revocation of attestation
+ * @dev This contract is abstract and provides utility functions to store attestations and revocations.
+ */
+abstract contract AttestationManager is IRegistry, ModuleManager, SchemaManager, TrustManager {
+    using StubLib for *;
+    using AttestationLib for AttestationDataRef;
+    using AttestationLib for AttestationRequest;
+    using AttestationLib for AttestationRequest[];
+    using AttestationLib for address;
+    using ModuleTypeLib for ModuleType[];
+
+    mapping(address module => mapping(address attester => AttestationRecord attestation)) internal
+        _moduleToAttesterToAttestations;
+
+    /*´:°•.°+.*•´.*:˚.°*.˚•´.°:°•.°•.*•´.*:˚.°*.˚•´.°:°•.°+.*•´.*:*/
+    /*                      Attestation                           */
+    /*.•°:°.´+˚.*°.˚:*.´•*.+°.•°:´*.´•*.•°.•°:°.´:•˚°.*°.˚:*.´+°.•*/
+
+    /**
+     * Processes an attestation request and stores the attestation in the registry.
+     * If the attestation was made for a module that was not registered, the function will revert.
+     * function will get the external Schema Validator for the supplied SchemaUID
+     *         and call it, if an external IExternalSchemaValidator was set
+     * function will get the external IExternalResolver for the module - that the attestation is for
+     *        and call it, if an external Resolver was set
+     */
+    function _attest(
+        address attester,
+        SchemaUID schemaUID,
+        AttestationRequest calldata request
+    )
+        internal
+    {
+        (AttestationRecord memory record, ResolverUID resolverUID) =
+            _storeAttestation({ schemaUID: schemaUID, attester: attester, request: request });
+
+        record.requireExternalSchemaValidation({ schema: schemas[schemaUID] });
+        record.requireExternalResolverOnAttestation({ resolver: resolvers[resolverUID] });
+    }
+
+    /**
+     * Processes an array of attestation requests  and stores the attestations in the registry.
+     * If the attestation was made for a module that was not registered, the function will revert.
+     * function will get the external Schema Validator for the supplied SchemaUID
+     *         and call it, if an external IExternalSchemaValidator was set
+     * function will get the external IExternalResolver for the module - that the attestation is for
+     *        and call it, if an external Resolver was set
+     */
+    function _attest(
+        address attester,
+        SchemaUID schemaUID,
+        AttestationRequest[] calldata requests
+    )
+        internal
+    {
+        uint256 length = requests.length;
+        AttestationRecord[] memory records = new AttestationRecord[](length);
+        // loop will check that the batched attestation is made ONLY for the same resolver
```

Comment:
could it make sense to allow different resolvers too or would this just be too much overhead in the base case

Reply:
probably possible, but would mean that we cant really use the "batched" call on the resolver, but use multiple calls for each record.

so im not sure if this is actually an upside, vs making multicall to the registry